Examples/Detectors/MagneticField/src/MagneticFieldOptions.cpp

```diff
@@ -175,6 +198,39 @@ ActsExamples::Options::readMagneticField(const Variables& vars) {
     }
   }
 
+  // third option: create a solenoid field
+  if (vars["bf-solenoid-mag-tesla"].as<double>() > 0) {
+    // Construct a solenoid field
+    Acts::SolenoidBField::Config solenoidConfig;
+    solenoidConfig.length =
+        vars["bf-solenoid-length"].as<double>() * Acts::UnitConstants::mm;
+    solenoidConfig.radius =
+        vars["bf-solenoid-radius"].as<double>() * Acts::UnitConstants::mm;
+    solenoidConfig.nCoils = vars["bf-solenoid-ncoils"].as<size_t>();
+    solenoidConfig.bMagCenter =
+        vars["bf-solenoid-mag-tesla"].as<double>() * Acts::UnitConstants::T;
+    ACTS_INFO("Use solenoid magnetic field with magnitude "
+              << solenoidConfig.bMagCenter / Acts::UnitConstants::T
+              << " Tesla at the center.");
+    const auto solenoidField = Acts::SolenoidBField(solenoidConfig);
+    // The parameters for creating a field map
+    auto getRange = [&](const char* name, auto unit, auto& lower, auto& upper) {
+      auto interval = vars[name].as<Options::Interval>();
+      lower = interval.lower.value() * unit;
+      upper = interval.upper.value() * unit;
+    };
+    std::pair<double, double> rlim, zlim;
+    getRange("bf-solenoid-map-rlim", Acts::UnitConstants::mm, rlim.first,
+             rlim.second);
+    getRange("bf-solenoid-map-zlim", Acts::UnitConstants::mm, zlim.first,
+             zlim.second);
+    const auto nbins = vars["bf-solenoid-map-nbins"].as<Reals<2>>();
+    auto mapper = Acts::solenoidFieldMapper(rlim, zlim, {nbins[0], nbins[1]},
+                                            solenoidField);
+    InterpolatedMagneticField2::Config cfg(std::move(mapper));
+    return std::make_shared<InterpolatedMagneticField2>(std::move(cfg));
```

Comment:
in principle I approve of zealous programming, such as moving everything that isn't used. I also see it's absolutely consistent with the rest of the code (so just leave it as is), I just want to mention that variables that aren't mentioned again are optimized out for optimization >= O1, so moving is not actually doing anything here.

Reply:
OK. Thank you for pointign that out! We should keep this in mind when writing new code.

Reply:
If it doesn't produce a warning, it most likely doesn't hurt doing the extra move.

Reply:
It certainly doesn't hurt, it's just making it a bit more complicated than necessary. I'm guilty of doing the same in the past. That's why I also didn't request it changed.